src/postprocessors/BetaEffective.C

```diff
@@ -0,0 +1,64 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2021 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "BetaEffective.h"
+
+#include "OpenMCProblemBase.h"
+
+registerMooseObject("CardinalApp", BetaEffective);
+
+InputParameters
+BetaEffective::validParams()
+{
+  auto params = GeneralPostprocessor::validParams();
+  params += OpenMCBase::validParams();
+  params.addClassDescription(
+      "A post-processor which computes and returns the kinetics parameter $\\beta_{eff}$.");
+
+  return params;
+}
+
+BetaEffective::BetaEffective(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters), OpenMCBase(this, parameters)
+{
+  if (openmc::settings::run_mode != openmc::RunMode::EIGENVALUE)
```

Comment:
Can this error check be removed? If on line 44 we are requiring OpenMC to have kinetics params computed, then I think we can rely on the eigenvalue mode check in `OpenMCCellAverageProblem` to reduce a bit of code duplication? (and test coverage)